iconservice/prep/data/prep_container.py

```diff
@@ -191,10 +191,19 @@ def get_by_address(self, address: 'Address') -> Optional['PRep']:
     def get_preps(self, start_index: int, size: int) -> List['PRep']:
         """Returns active P-Reps ranging from start_index to start_index + size - 1
 
-        :return: P-Rep list
+        :return: Active P-Rep list
         """
         return self._active_prep_list[start_index:start_index + size]
 
+    def get_inactive_preps(self):
+        """Returns generator of inactive P-Reps
+
+        :return: generator of inactive P-Reps
+        """
+        for prep in self._prep_dict.values():
+            if prep not in self._active_prep_list:
```

Comment:
Its time complexity looks n^2 (inefficient)
Use prep.status instead of self._active_prep_list

Reply:
Solved it.